x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/mapper/expressiondsl/FieldExpression.java

```diff
@@ -148,6 +148,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             return builder.value(value);
         }
 
+        @Override
+        public String toString() {
+            return Objects.toString(value);
```

Comment:
Yes, multi-valued expression are a `List<FieldValue>` rather than a `FieldValue` over a list/array.
Given that, I don't think it's worth trying to detect an array here. If there was an easily available method for `maybeDeepToString(Object)` we could call it, but I don't know of one in our codebase.